inst/rmarkdown/lua/number-sections.lua

```diff
@@ -32,37 +36,112 @@ local section_number_table = {0, 0, 0, 0, 0, 0, 0, 0, 0}
 local n_section_number_table = #section_number_table
 local previous_header_level = 0
 local separator = pandoc.Space()
+local identifiers = {}
 if FORMAT == "docx" then -- to be consistent with Pandoc >= 2.10.1
   separator = pandoc.Str("\t")
 end
 
-function Header(elem)
-  -- If unnumbered
-  if (elem.classes:find("unnumbered")) then
-    return elem
+
+function is_preprocess(meta)
+  -- Check if the filter is run in preprocessing step
+  -- This should only be done for markdown document conversion
+  preprocess = meta['preprocess_number_sections']
+  return nil
+end
+
+
+function fix_link_id(link)
+  -- Some markdown formats needs fixing on there link to numbered sections
+  -- idenfiers is filled during processing when a pre processing has been run
+  if not preprocess and #identifiers and identifiers[link.target] then
+    link.target = identifiers[link.target]
+    return link
+  end
+  -- do nothing in all other cases
+  return nil
+end
+
+
+function number_sections(header)
+
+  ----------------------------
+  -- specific handling when pre-processing has been done
+  ----------------------------
+
+  -- This will only run on second conversion when a preprocessing has be run before
+  -- to add recorded attributes set by us as Span attribute in first processing.
+  if not preprocess and #header.content and (header.content[1].t == 'Span') then
+      -- Check attributes
+      local span_content = header.content[1]
+      local old_id = span_content.attributes['data-rmarkdown-temporarily-recorded-id']
+      -- If found, this is our Span added in pre processing step
+      if old_id then
+        identifiers['#' .. old_id] = '#' .. header.identifier
+        -- remove span
+        header.content:remove(1)
```

Comment:
@cderv 
I think `remove` is added since 2.9.2 with the following commit:
https://github.com/jgm/pandoc/commit/4ede11a75f37bb35d384fc9ce95231671e18acf3

If you want to support older versions, you need some workaround.

```lua
table_copy = {}
for i, v in ipairs(table_original) do
  if i > 1 then
    table.insert(table_copy, v)
  end
end
```

Reply:
I have started here https://gist.github.com/cderv/de33e6171ff181d7cce7e7630a8b9935 but should probably move in Wiki to offer contribution to others.